app/models/payment.rb

```diff
@@ -28,6 +28,8 @@ class Payment < ApplicationRecord
 
   validates :status, inclusion: ORDER_PAYMENT_STATUS.values
 
+  scope :completed, -> { where(status: ORDER_PAYMENT_STATUS['successful']) }
+
```

Comment:
You rely on this scope to be ordered for e.g. the "most recent payment" and the "membership expiry date". So I think there should be an order clause here.

Reply:
I was relying on default ordering by primary ID (asc), but I'm not sure that Rails will guarantee that ID is always increasing consistent with `created_at`.  That is, this will work (I'm pretty sure) with Postgrsql, but not necessarily with another DB.

In any case, I will apply ordering in the User model rather than here, as I want to defer ordering to the point of use - which may require different ordering depending on context.